src/com/sap/piper/Utils.groovy

```diff
@@ -175,3 +175,13 @@ static List appendParameterToStringList(List list, Map parameters, String paramN
     }
     return result
 }
+
+static String getStageName(Script script, Map parameters, String technicalStageName) {
```

Comment:
What about handing the step in here, or passing in the other two stage names as string as well?

```suggestion
static String getStageName(Script script, Map parameters, Script step) {
```